Edit steps (#16)
aws-samples/how-to-write-more-correct-software-workshop: instructions/steps.md

Dafny is still complaining.
Let's think about this.
`ParseAwsKmsResource` does not have any `ensures` on it.

> Dafny know now that it is possible for `ParseAwsKmsResource` to succeed or fail,

Reviewer: ```suggestion
Dafny knows now that it is possible for `ParseAwsKmsResource` to succeed or fail,
```

Let's go over some cool things about this moment.

First, we have added constraints to a method *before* implementing it.

> If you tried to compile this, Dafny would complain
> since there are no steps to execute.
> But Dafny has proved that *any* steps
> that result in these postconditions could work.

Reviewer: ```suggestion
If you tried to compile this, Dafny would complain
since there is no implementation to execute.
But Dafny has proved that *any* implementation
that results in these postconditions could work.
```

Or you could go with "body". I'd avoid "steps" both because it implies sequential statements, and you're already using Step # to program your workshop participants. :)

we are done with this section!

Finally, take a look at the requirement "The resource id MUST be a non-empty string"
and see if you can work out why this is indeed enforced.[^non-empty]

[^non-empty]: We used `<=` as a way to express "starts with".
    In this case we are using `<`.
    This means that the `arnResource` string can not be equal
    but MUST have additional characters after the `/`.
    Since we have proved
    `arnResource == result.value.resourceType + "/" + result.value.value`
    if there are characters after the `/` then the resource id

>     MUST be non-empty.

Reviewer: Nice
